docs/user-guide/slicing.ipynb

```diff
@@ -217,6 +227,203 @@
     "d['x', 1:2]['a'] == d['a']['x', 1:2]\n",
     "d['x', 1:2]['a'].coords['x'] == d.coords['x']['x', 1:2]"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Label based indexing\n",
+    "\n",
+    "Data in a [dataset](../generated/scipp.Dataset.rst#scipp.Dataset) or [data array](../generated/scipp.DataArray.rst#scipp.DataArray) can be sliced by the coordinate value. This is similar to pandas [pandas.DataFrame.loc](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.loc.html).\n",
+    "Label based indexing with a [variable](../generated/scipp.Variable.rst#scipp.Variable)  or an [variable](../generated/scipp.Variable.rst#scipp.Variable) range is made via `__getitem__` and `__setitem__` with a dimension label as first argument.\n",
+    "This is available for data arrays and datasets, as well as items of a dataset, but not for variables as coordinates are necessary for this to work on any data structure.\n",
+    "In all cases a *view* is returned, i.e., just like when slicing a [numpy.ndarray](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.html#numpy.ndarray) no copy is performed.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lets create a simple data array. It has dimension coordinates in `x` and `y`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "da = sc.DataArray(data = sc.Variable(dims=['x', 'y'], values=np.arange(8.0).reshape(2, 4)),\n",
+    "    coords={\n",
+    "        'x': sc.Variable(['x'], values=np.flip(np.arange(2.0)), unit=sc.units.m),\n",
+    "        'y': sc.Variable(['y'], values=np.arange(4.0), unit=sc.units.m)})\n",
+    "sc.show(da)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lets say we wish to slice the data array where y coord is exactly at the point 1.0 meters expressed as a variable. You will notice that the data item in the output have no y coordinate dependency after the point slicing operation."
```

Comment:
I would avoid introducing point slicing based on float-coords, since in practice this will not work. Use an int or string coord instead?

Reply:
Agreed.